docs/src/main/paradox/google-cloud-bigquery.md

```diff
@@ -0,0 +1,101 @@
+
+For the rawest representation there is a "csvStyle" source built in. 
+This will return a header (field names), and the fields as a list of Strings.
+
+Scala
+: @@snip [snip](/google-cloud-bigquery/src/test/scala/docs/scaladsl/GoogleBigQuerySourceDoc.scala) { #csv-style }
+
+Java
+: @@snip [snip](/google-cloud-bigquery/src/test/java/docs/javadsl/GoogleBigQuerySourceDoc.java) { #csv-style }
+
+There is a more sophisticated way to get data from a database.
+If you want to get a stream of classes, you can add your converter function too.
+
+Scala
+: @@snip [snip](/google-cloud-bigquery/src/test/scala/docs/scaladsl/GoogleBigQuerySourceDoc.scala) { #run-query }
+
+Java
+: @@snip [snip](/google-cloud-bigquery/src/test/java/docs/javadsl/GoogleBigQuerySourceDoc.java) { #run-query }
+
+If you want to use the built in paging implementation, or you have some specific needs you can call the raw api.
+The next example shows how you can access [dryRun](https://cloud.google.com/bigquery/query-plan-explanation) data with the raw api and helpers.
+
+Scala
+: @@snip [snip](/google-cloud-bigquery/src/test/scala/docs/scaladsl/GoogleBigQuerySourceDoc.scala) { #dry-run }
+
+Java
+: @@snip [snip](/google-cloud-bigquery/src/test/java/docs/javadsl/GoogleBigQuerySourceDoc.java) { #dry-run }
+
+### Config
+
+The config will handle your session, and your service-token. (See the creation code above.)
+
+If you create multiple requests to the same source (likely to happen) you should create it once and try to reuse it.
+If you call multiple bigquery sources (not likely to happen) it is worth to cache the configs, so you can save a lot of unneeded authorization requests.
+
+### Cancel on timeout
+
+All of the provided functionality can fire a callback when the **downstream** signals a stop.
+This is useful if you want to implement some timeout in the downstream, and try to lower your costs with stopping the longrunning jobs.
+(Google doesn't provide any insurance about cost reduction, but at least we could try. [Read this for more information.](https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs/cancel))
+
+Scala
+: You can use the build in @scala[@scaladoc[BigQueryCallbacks](akka.stream.alpakka.googlecloud.bigquery.scaladsl.BigQueryCallbacks$).]
+
+Java
+: You can use the build in @java[@scaladoc[BigQueryCallbacks](akka.stream.alpakka.googlecloud.bigquery.javadsl.BigQueryCallbacks$).]
+
+### Parsers
+
+The parser function is a `JsObject => Option[T]` function. 
+This is needed because there is a possibility, the response not to contain any data. In this case we need to retry the request with some delay.
+Your parser function needs to be bulletproof and the code in the examples represents the happy path.
+If you return `None` in every error case; your stream will be polling forever!
+
+## Running the examples
+
+To run the example code you will need to configure a project, create/init tables in google-bigquery and provide your own credentials.
```

Comment:
What example are you referring to? This section could use some more ifnormation on how to config and run the example.

Reply:
Changed the documentation in order to provide a link to the e2e example